### ISSUE TITLE
[steamshovel] bubbles-artist does not refresh on missing key (Trac #1402)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1394]
When cycling through frames and a key is visualized with an artist (hit-bubbles), which is not present in a following frame, the visual information of the previous frame remains active (bubbles show, which there is actually no info for in the frame).

forcing the window to re-draw for example by cycling desktops fixes the visualized information. so it seems this is a refresh issue.

proposed resolution: call the 'refresh'  every time a frame is changed; I guess currently this is called by the artist themselves, which try to update only if their key to operate on is in fact present in the frame.

Migrated from https://code.icecube.wisc.edu/ticket/1402
```json
{
    "status": "closed", 
    "changetime": "2016-03-18T21:14:08", 
    "description": "When cycling through frames and a key is visualized with an artist (hit-bubbles), which is not present in a following frame, the visual information of the previous frame remains active (bubbles show, which there is actually no info for in the frame).\n\nforcing the window to re-draw for example by cycling desktops fixes the visualized information. so it seems this is a refresh issue.\n\nproposed resolution: call the 'refresh'  every time a frame is changed; I guess currently this is called by the artist themselves, which try to update only if their key to operate on is in fact present in the frame.", 
    "reporter": "mzoll", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1458335648518728", 
    "component": "combo core", 
    "summary": "[steamshovel] bubbles-artist does not refresh on missing key", 
    "priority": "normal", 
    "keywords": "steamshovel refresh", 
    "time": "2015-10-16T19:22:06", 
    "milestone": "", 
    "owner": "hdembinski", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20151016T22:05:30: 
* hdembinski commented:

> Thank you for reporting this. That's weird. It should work in principle, so there must be a bug in the logic that detects when an update is necessary. I will look into it.


* hdembinski changed status from "new" to "accepted"

**Comment by icecube-trac** [comment]
Trac update at 20151029T10:32:21: olivas changed milestone from "November 2015" to "December 2015"

**Comment by icecube-trac** [comment]
Trac update at 20151104T15:42:15: hdembinski commented:

> The proposed solution cannot be that simple, unfortunately, since there are artists whose input remains valid from frame to frame, e.g. the Detector artist. Steve has some complicated logic at work, which checks an artist has to be updated or not. Avoiding unnecessary computations is a good thing, so we should keep that, but the logic has to be fixed where it is broken.

**Comment by icecube-trac** [comment]
Trac update at 20151104T16:28:11: 
* hdembinski changed resolution from "" to "fixed"
* hdembinski changed status from "accepted" to "closed"
* hdembinski commented:

> In changeset 139170:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="139170"
> fixing missing update, closes #1394
> ```

**Comment by icecube-trac** [comment]
Trac update at 20160318T21:14:08: 
* olivas commented:

> Milestone December 2015 deleted


* olivas changed milestone from "December 2015" to ""
